bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -72,28 +74,47 @@ impl S2NHarness {
         }
     }
 
-    fn create_config(mode: Mode, crypto_config: &CryptoConfig) -> Result<Config, Box<dyn Error>> {
-        let security_policy = match (&crypto_config.cipher_suite, &crypto_config.ec_group) {
+    fn create_config(
+        mode: Mode,
+        crypto_config: CryptoConfig,
+        handshake_type: HandshakeType,
+    ) -> Result<Config, Box<dyn Error>> {
+        let security_policy = match (crypto_config.cipher_suite, crypto_config.ec_group) {
             (CipherSuite::AES_128_GCM_SHA256, ECGroup::SECP256R1) => "20230317",
             (CipherSuite::AES_256_GCM_SHA384, ECGroup::SECP256R1) => "20190802",
             (CipherSuite::AES_128_GCM_SHA256, ECGroup::X25519) => "default_tls13",
             (CipherSuite::AES_256_GCM_SHA384, ECGroup::X25519) => "20190801",
         };
 
         let mut builder = Builder::new();
-        builder.set_security_policy(&Policy::from_version(security_policy)?)?;
-
-        match mode {
-            Mode::Server => builder.load_pem(
-                read_to_bytes(SERVER_CERT_CHAIN_PATH).as_slice(),
-                read_to_bytes(SERVER_KEY_PATH).as_slice(),
-            )?,
-            Mode::Client => builder
+        builder
+            .set_security_policy(&Policy::from_version(security_policy)?)?
+            .wipe_trust_store()?
+            .set_client_auth_type(match handshake_type {
+                HandshakeType::Full => ClientAuthType::None,
+                HandshakeType::mTLS => ClientAuthType::Required,
+            })?;
+
+        // add CA cert if needed
+        if handshake_type == HandshakeType::mTLS || mode == Mode::Client {
```

Comment:
This configuration logic(as well as on line 108) is starting to get very difficult to read. Can you fix it up so that it's easier to see what's going on?

Reply:
I'll separate out the function into `create_common_config()`, `create_client_config()`, and `create_server_config()`.